Update to dbsync_deamon

JanCaha opened the pull request:
Fixes #97 
Fixes #94 

- Allows dbsync_deamon.py to be run in single run mode, where only one run is performed.
- dbsync.py does not run as a CLI anymore
- better handling of exceptions (with exit(1))

wonder-sk reviewed (commented):
> Should also solve https://github.com/MerginMaps/mergin-db-sync/issues/91 if dbsync.py will not be runned directly (the ability to run it could be also removed as part of this PR).

yeah feel free to remove the option to run dbsync.py directly as a CLI tool

> Should the error also end python using exit(1), so it does not end as success?

That would make sense to me. Just in case of pull/push within db sync daemon, I would not exit, as it could be just a temporary network/server issue which should not stop the daemon altogether...

JanCaha commented:
- removed ability to run dbsync.py as CLI tool
- will exit with error code unless in while loop
- also fixed #94 - now skip_tables can be empty, empty list, list or string and the `get_ignored_tables()` will return correct representation - this includes couple of new test
